test/multi_process_test.rb

```diff
@@ -28,10 +28,89 @@
     Resque.schedule = schedules
     counts << Resque.schedule.size
 
-    threads.each(&:join)
+    counts += get_results_from_children(pids)
 
     counts.each_with_index do |c, i|
       assert_equal schedule_count, c, "schedule count is incorrect (c: #{i})"
     end
   end
+
+  test 'concurrent shutdowns and startups do not corrupt the schedule' do
+    counts = []
+    children = []
+
+    processes = 100
```

Comment:
Probably not :P I'll try a lower number
On Mar 30, 2016 8:43 AM, "Florian Weingarten" <notifications@github.com>
wrote:

> In test/multi_process_test.rb
> <https://github.com/Shopify/resque-scheduler/pull/3#discussion_r57881123>:
>
> >
> >      counts.each_with_index do |c, i|
> >        assert_equal schedule_count, c, "schedule count is incorrect (c: #{i})"
> >      end
> >    end
> > +
> > +  test 'concurrent shutdowns and startups do not corrupt the schedule' do
> > +    counts = []
> > +    children = []
> > +
> > +    processes = 100
>
> This seems like a lot. Do we need this many in order to make the test fail
> consistently without the patch?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/resque-scheduler/pull/3/files/ee7301e9458b97bbec2a26804e30894319a15d90#r57881123>
>